kv/dist_sender.go

```diff
@@ -353,8 +351,7 @@ func (ds *DistSender) sendRPC(trace *tracer.Trace, rangeID proto.RangeID, replic
 	var addrs []net.Addr
 	replicaMap := map[string]*proto.Replica{}
 	for i := range replicas {
```

Comment:
i believe that makes a shallow copy which is reused for each element, which makes taking the address of `replicas[i].Replica` below not equivalent